game-app/game-core/src/main/java/games/strategy/triplea/delegate/TransportTracker.java

```diff
@@ -221,11 +221,11 @@ public static Collection<Unit> getUnitsLoadedOnAlliedTransportsThisTurn(
   /** Detects if a unit has unloaded units in a previous game phase. */
   public static boolean hasTransportUnloadedInPreviousPhase(final Unit transport) {
     final Collection<Unit> unloaded = transport.getUnloaded();
+    boolean nonCombat = GameStepPropertiesHelper.isNonCombatMove(transport.getData(), true);
     // See if transport has unloaded anywhere yet
     for (final Unit unit : unloaded) {
       // cannot unload in two different phases
-      if (GameStepPropertiesHelper.isNonCombatMove(transport.getData(), true)
-          && unit.getWasUnloadedInCombatPhase()) {
+      if (nonCombat && unit.getWasUnloadedInCombatPhase()) {
```

Comment:
I would even go one step further. Why loop over the whole unloaded collection when nonCombat is false, when you can just return early.

Reply:
Even better